tests/test_bundle_tls_provider.py

```diff
@@ -27,6 +27,7 @@ def lightkube_client() -> lightkube.Client:
 
 
 @pytest.mark.abort_on_fail
+@pytest.mark.skip_if_deployed
```

Comment:
Usually we don't use `@pytest.mark.skip_if_deployed` in the CI, as nothing should be deployed beforehand, otherwise this can be considered an error. Did you leave them by mistake?

Reply:
This is used only if in pytest-operator you call testing with `--no-deploy`, so it's really useful if you want to run test multiple time locally, as I was doing. We can remove it if you think so, but I do not see any harm to leave it there. 

I think that a lot of CKF tests using decorator `pytest.mark.abort_on_fail` wrongly, since it's meant to be used only once for part which is responsible for deployment. At the same time it should be used with skip_if_deployed decorator so you can easily run tests multiple time (with `--no-deploy`) without need to run test from scratch (create new model, build charm, deploy it and others, etc.). For example I do not understand why test below `test_tls_configuration`, is marked with `abort_on_fail`? It's only one test.

Reply:
Would it make sense then to have a small refactoring PR where we remove the decorators that do not make sense and add the ones that can help the dev experience a bit better?

I can definitely see the value of `@pytest.mark.skip_if_deployed` when running the tests locally, despite this test file being used mostly in the CI.